### PR TITLE
Intellij 15.0.3 code inspection done and obvious/easy fixes applied

### PR DESCRIPTION
**Pull request by guidomedina**
I have ran the IntelliJ code inspections and the following obvious/easy fixes have been applied only where it made sense, if on doubt it was skipped:

* redundant return variables.
* if-statement simplifications.
* unnecessary imports removed.
* unnecessary boxing/unboxing removed.
* unnecessary type-casting removed.
* some String concatenation within StringBuilder replaced by StringBuilder append.

The fixes for their corresponding inspections were carefully applied after humanly checking that IntelliJ would not introduce a side-effect.

This branch will result result on a slightly better GC before JIT optimizes, no side-effect or functionality or feature has been modified or added.

**Signed the Netty CLA**

### CODE REVIEW

**Comment by Scottmitch** [comment]
@normanmaurer - 3.x = assign to you :wink:

**Comment by guidomedina** [comment]
I tried to do the same thing for branch 4.0 but realized you are avoiding using iterators and boxing/unboxing purposely so I wonder now if my branch will make any sense to you, except for the if simplifications.

Are you 100% sure all collections you are iterating will give you a Big O(n) = 1 for ```List.get(index)``` call?

File: src/main/java/org/jboss/netty/handler/ssl/SslHandler.java [hunk 1]
```diff
@@ -1655,9 +1655,8 @@ public void run() {
 
                 if (futures != null) {
                     final ClosedChannelException cause = new ClosedChannelException();
-                    final int size = futures.size();
-                    for (int i = 0; i < size; i ++) {
-                        futures.get(i).setFailure(cause);
+                    for (ChannelFuture future : futures) {
```
Comment (huguesb) [inline src/main/java/org/jboss/netty/handler/ssl/SslHandler.java]: This implicitly creates an Iterator, which will generate garbage to be collected.

Comment (slandelle) [inline src/main/java/org/jboss/netty/handler/ssl/SslHandler.java]: JIT's escape analysis should be able to get rid of it once it kicks in, but yes, first 10.000 iterations will get a penalty.

**Comment by huguesb** [comment]
Afaict most of these changes are just relying on the compiler to do implicitly what was done explicitly. It makes for slightly more readable code but it is unlikely to affect GC or CPU usage.

File: src/main/java/org/jboss/netty/handler/ssl/SslHandler.java (hunk 1 above)
Comment (guidomedina) [inline src/main/java/org/jboss/netty/handler/ssl/SslHandler.java]: I can undo this particular change and force push the branch, I know there isn't probably any gain with the proposed changes but at least the code is more readable, but indeed not sure if is worth the effort, let me know.

**Comment by guidomedina** [comment]
I just re-pushed with implicit iterator removed.

File: src/main/java/org/jboss/netty/handler/ssl/SslHandler.java (hunk 1 above)
Comment (guidomedina) [inline src/main/java/org/jboss/netty/handler/ssl/SslHandler.java]: I changed that and removed ```size``` variable and left ```futures.size()```, the rest I kept the same. I force pushed so it is the same set of changes but no iterator.

**Comment by guidomedina** [comment]
I removed the implicit iterator and ```size``` variable and replaced by ```futures.size()```, the rest I kept the same. I force pushed so it is the same set of changes without iterator.

File: src/main/java/org/jboss/netty/channel/group/CombinedIterator.java [hunk 2]
```diff
@@ -55,8 +55,7 @@ public boolean hasNext() {
     public E next() {
         for (;;) {
             try {
-                E e = currentIterator.next();
```
Comment (johnou) [inline src/main/java/org/jboss/netty/channel/group/CombinedIterator.java]: if next is volatile it must be assigned to a local variable first.

**Comment by johnou** [comment]
-1 for if-statement simplifications. Some are ok but as @huguesb mentioned I doubt there would be any performance improvement and the readability of the code is not as easy as before. Another possible problem with merging multiple object field checks on one line is debugging NPEs.

**Comment by guidomedina** [comment]
Short-circuit boolean expressions are well known to be optimized away and for debugging you usually break point at the target line and see the current variable values, some if-statements don't even make much sense TBH

**Comment by guidomedina** [comment]
Netty 3.x is pretty much at its EOL which makes safer to format code and stuff.

**Comment by guidomedina** [comment]
Re-based from current 3.10 to make diff easier though it is unlikely this branch is going to be merged.

**Comment by guidomedina** [comment]
Re-based from current 3.10 to make diff easier and corrected an iterator I thought I had changed back to old way.

File: src/main/java/org/jboss/netty/handler/ssl/SslHandler.java [hunk 3]
```diff
@@ -1656,8 +1656,7 @@ public void run() {
 
                 if (futures != null) {
                     final ClosedChannelException cause = new ClosedChannelException();
-                    final int size = futures.size();
-                    for (int i = 0; i < size; i ++) {
+                    for (int i = 0; i < futures.size(); i++) {
```
Comment (johnou) [inline src/main/java/org/jboss/netty/handler/ssl/SslHandler.java]: Better to assign futures.size() to a variable first so it's not calculated every iteration.

Comment (guidomedina) [inline src/main/java/org/jboss/netty/handler/ssl/SslHandler.java]: is not really calculated, it is the variable length of an array, but OK, I just restored that file.

Comment (guidomedina) [inline src/main/java/org/jboss/netty/handler/ssl/SslHandler.java]: sorry I meant the internal variable size of ```ArrayList```

**Pull request closed without merging**